tests/erc721/erc721_test.go

```diff
@@ -162,6 +166,105 @@ func TestOpenSeaProxyApproval(t *testing.T) {
 	}
 }
 
+func TestOpenSeaProxyPreApproval(t *testing.T) {
+
+	tests := []struct {
+		name            string
+		hasProxy        bool
+		wantPreApproved bool
+	}{
+		{
+			name:            "without proxy on mint",
```

Comment:
"on transfer" perhaps? Because that's the general case, and we're using mint as an easy example of a transfer.

Reply:
I think there is a bit of ambiguity around the term. 
- In the general case, there's no assumptions on `from` and `to`. So they would also include burns (i.e. `to = 0`), for which we would expect a different behavior.
- In the more specific case, one would assume `from != to` with both being non-zero. So they wouldn't include mints.

Reply:
So I'd rather include two more tests for special transfers and burns.